src/Semantics/EqInterpRefl.dfy

```diff
@@ -0,0 +1,97 @@
+include "../Interop/CSharpDafnyASTModel.dfy"
+include "../Interop/CSharpInterop.dfy"
+include "../Interop/CSharpDafnyInterop.dfy"
+include "../Interop/CSharpDafnyASTInterop.dfy"
+include "../Utils/Library.dfy"
+include "../Utils/StrTree.dfy"
+include "Interp.dfy"
+include "Equiv.dfy"
+include "ExprInduction.dfy"
+include "EqInterpScopes.dfy"
+
+module Bootstrap.Semantics.EqInterpRefl {
+  // This module provides lemmas which state that evaluating an expression with equivalent contexts
+  // (in the sense of ``EqState``) leads to equivalent results.
+  //
+  // We simply reuse the results from `EqInterpScopes`, which are stricly more general.
+
+  import opened AST.Syntax
+  import opened Utils.Lib
+  import opened AST.Predicates
+  import opened Interp
+  import opened Values
+  import opened Equiv
+  import opened Utils.Lib.Datatypes
+  import opened EqInterpScopes
+
+  type Expr = Interp.Expr
+
+  lemma InterpBlock_Exprs_EqRefl(es: seq<Expr>, env: Environment, ctx: State, ctx': State)
+    // TODO(SMH): for some reason, using ``Seq_All`` makes some proofs fail. The weird thing is
+    // that I can then prove `Seq_All(SupportsInterp, es)` in an assertion just before the call
+    // to the lemma, but the lemma precondition keeps failing.
+    requires forall e | e in es :: SupportsInterp(e)
```

Comment:
Does it work with `Seq.All`?  If not, can we minimize, since we now have three separate idioms to say `all`?

Reply:
I tried using `Seq.All` at some point, but it led to several failures. I will minimize